e2e-tests/development-runtime/cypress/integration/redirect/redirect.js

```diff
@@ -0,0 +1,17 @@
+describe(`redirect`, () => {
+  beforeEach(() => {
+    cy.visit(`/`).waitForRouteChange()
+  })
+
+  it(`should redirect page to index page when there is no such page`, () => {
+    cy.visit(`/redirect-without-page`).waitForRouteChange()
+
+    cy.location(`pathname`).should(`equal`, `/`)
+  })
+
+  it(`should redirect page to index page even there is a such page`, () => {
+    cy.visit(`/redirect`).waitForRouteChange()
+
+    cy.location(`pathname`).should(`equal`, `/`)
```

Comment:
I assume in this case we will get `The route "${pathname}" matches both a page and a redirect; this is probably not intentional.` error in browser console? Wonder if we can assert that console entry

Reply:
added this in hacky way, resolving the comment